assets/pages/paper-subscription-landing/paperSubscriptionLandingPageReducer.js

```diff
@@ -0,0 +1,26 @@
+// @flow
+
+// ----- Imports ----- //
+
+import type { CommonState } from 'helpers/page/commonReducer';
+import { type PaperBillingPeriod } from 'helpers/subscriptions';
+import { getQueryParameter } from 'helpers/url';
+import { productPagePeriodFormReducerFor, type State as FormState } from 'components/productPage/productPagePeriodForm/productPagePeriodFormReducer';
+
+export type State = {
+  common: CommonState,
+  page: FormState<PaperBillingPeriod>,
+};
+
+
+// ----- Reducer ----- //
+
+const promoInUrl = getQueryParameter('promo');
+
+const initialPeriod: ?PaperBillingPeriod =
+  promoInUrl === 'sixday' || promoInUrl === 'weekend' || promoInUrl === 'sunday' || promoInUrl === 'month' ? promoInUrl : null;
```

Comment:
minor: would `['sixday', 'weekend', 'sunday', 'month'].contains(promoInUrl)` be clearer?

Reply:
it absolutely would be clearer but flow doesn't understand it 😝